packages/dds/tree/src/shared-tree/treeAlpha.ts

```diff
@@ -481,6 +513,72 @@ export const TreeAlpha: TreeAlpha = {
 		const parentSchema = treeNodeApi.schema(parent);
 		return getPropertyKeyFromStoredKey(parentSchema, storedKey);
 	},
+
+	child: (node: TreeNode, key: string | number): TreeNode | TreeLeafValue | undefined => {
+		const flexNode = getOrCreateInnerNode(node);
+		debugAssert(() => !flexNode.context.isDisposed() || "FlexTreeNode is disposed");
+
+		const schema = treeNodeApi.schema(node);
+		const storedKey = tryGetStoredKeyFromPropertyKey(schema, key);
+		if (storedKey === undefined) {
+			return undefined;
+		}
+
+		if (isArrayNodeSchema(schema)) {
+			const sequence = flexNode.getBoxed(EmptyKey);
+
+			const index =
+				typeof storedKey === "number"
+					? storedKey
+					: asIndex(storedKey, Number.POSITIVE_INFINITY);
+
+			if (index === undefined) {
+				return undefined;
+			}
+
+			const child = sequence.boxedAt(index);
+			if (child === undefined) {
+				return undefined;
+			}
+
+			return getOrCreateNodeFromInnerNode(child);
+		}
+
+		const field = flexNode.tryGetField(brand(String(storedKey)));
+		if (field !== undefined) {
+			return getTreeNodeForField(field);
+		}
+
+		return undefined;
+	},
+
+	children: (node: TreeNode): [string | number, TreeNode | TreeLeafValue][] => {
+		const flexNode = getOrCreateInnerNode(node);
+		debugAssert(() => !flexNode.context.isDisposed() || "FlexTreeNode is disposed");
+
+		const schema = treeNodeApi.schema(node);
+
+		const result: [string | number, TreeNode | TreeLeafValue][] = [];
+
+		if (isArrayNodeSchema(schema)) {
+			const sequence = flexNode.getBoxed(EmptyKey);
+
+			for (const childFlexNode of sequence.boxedIterator()) {
```

Comment:
using boxed iterator adds the overhead of allocating flex tree nodes for the leaves.

FlexTreeSequenceField.map would be more efficient.

Also for actually processing the unboxed flex tree node, maybe add a helper to src/simple-tree/getTreeNodeForField.ts with the

```
isFlexTreeNode(maybeContent)
			? getOrCreateNodeFromInnerNode(maybeContent)
			: maybeContent;
```
 portion, and dedupe that with the array node getter and getTreeNodeForField